src/testing/components.ts

```diff
@@ -36,18 +36,23 @@ export function cit(desc: string, testFn: ComponentTestCallback) {
 }
 
 // cms mocked in all component tests
-let mockCms: MockCmsService;
-export function currentCms() {
-  return mockCms;
+let mockHal: MockHalService;
+export function currentHal() {
+  return mockHal;
 }
 
 // create a component
 export function create(componentType: Type<any>, runInitialDetect = true) {
-  mockCms = new MockCmsService();
+  mockHal = new MockHalService();
   beforeEach(function() {
     this._publishComponent    = componentType;
     this._publishDeclarations = [componentType, StubRouterLinkDirective];
-    this._publishProviders    = [{provide: CmsService, useValue: mockCms}, {provide: CastleService, useValue: mockCms}];
+    this._publishProviders    = [
+      {provide: HalService, useValue: mockHal},
```

Comment:
We're just mocking out the HalService with the mock one.  This way, we can just use the actual Castle/Cms/Feeder services, and dependency injection will provide the MockHalService to them.